src/approaches/tamp_approach.py

```diff
@@ -61,3 +52,38 @@ def _get_current_predicates(self) -> Set[Predicate]:
         Defaults to initial predicates.
         """
         return self._initial_predicates
+
+
+def option_plan_to_policy(plan: Sequence[_Option]
+                          ) -> Callable[[State], Action]:
+    """Create a policy that executes the options in order.
+
+    We may want to move this out of here later, but I'm leaving
+    it for now, because it's annoying to import ApproachFailure
+    in utils.py, for example.
+
+    The logic for this is somewhat complicated because we want:
+    * If an option's termination and initiation conditions are
+      always true, we want the option to execute for one step.
+    * After the first step that the option is executed, it
+      should terminate as soon as it sees a state that is
+      terminal; it should not take one more action after.
+    """
+    queue = list(plan)  # Don't modify plan, just in case
+    initialized = False  # Special case first step
+    def _policy(state: State) -> Action:
+        nonlocal initialized
+        # On the very first state, check initiation condition, and
+        # take the action no matter what.
+        if not initialized:
+            if not queue:
+                raise ApproachFailure("Ran out of options in the plan!")
+            assert queue[0].initiable(state), "Unsound option plan"
+            initialized = True
+        elif queue[0].terminal(state):
+            queue.pop(0)
```

Comment:
FYI from my interview prep lol, pop(0) is an O(n) operation, so if we were actually doing this correctly we'd want to reverse the queue, but it doesn't matter since our plans are all miniscule

Reply:
huh! good to know